File: 1422. Maximum Score After Splitting a String.py
class Solution:
    # O(n) time and O(1) space complexity
    def maxScore(self, s: str) -> int:
        ones = 0
        for digit in s:
            ones += int(digit)

        score = zeroes = 0
        for digit in s[:-1]:
            if digit == '1':
                ones -= 1
            else:
                zeroes += 1
            score = max(score, zeroes + ones)

        return score

File: 1422. Maximum Score After Splitting a String.cpp
class Solution {
public:
    // O(n) time and O(1) space complexity
    int maxScore(string s) {
        int ones = 0;
        for (char &digit : s) {
            if (digit == '1') {
                ones++;
            };
        };

        int score = 0, zeroes = 0;

        for (int i = 0; i < s.size() - 1; i++) {
            if (s[i] == '1') {
                ones--;
            } else {
                zeroes++;
            };

            score = max(score, ones + zeroes);
        };

        return score;
    };
};
